### PR TITLE
feature: added ability to prefix utility classes and variables

### PR DESCRIPTION
**Pull request by bytelabsco**
Also added documentation to readme.md

### CODE REVIEW

**Comment by bytelabsco** [comment]
closes #25

**Comment by Andy-set-studio** [comment]
This is looking great! I'm right in a crunch on client work so want to give this a proper review when I'm done (hopefully early november), to give it the right justice it deserves. 

I'll leave some passing comments now, but thank you so much for this. It's a really good start!!

File: src/_default-config.scss [hunk 1]
```diff
@@ -34,6 +34,11 @@ $gorko-colors: (
 /// to enabling/disabling pre-built components/utilities.
 ///
 $gorko-config: (
+  'namespace': (
+    'prefix': '',
+    'classes': true,
+    'vars': false
+  ),
```
Comment (Andy-set-studio) [inline src/_default-config.scss]: Once concern I have here is that we need to support people updating Gorko and not being aware of this feature. Is there an accommodation for this config section not existing?

File: src/generator/workers/_get-namespace.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+///
+/// @param {string} $namespace-type - Which type of object needs namespaced.  One of 'classes' or 'vars'
+///
+@function get-namespace($namespace-type) {
+
+    $namespace-defaults: (
+        'prefix': '',
+        'classes': true,
+        'vars': false
+    );
```
Comment (Andy-set-studio) [inline src/generator/workers/_get-namespace.scss]: Ah this answers my above comment about what happens when the config doesn't exist!

**Comment by bytelabsco** [comment]
Just wanted to touch base and see if there's anything I can do to help.

**Comment by Andy-set-studio** [comment]
Ah sorry, @bytelabsco. I'll review it now :)

File: readme.md [hunk 3]
```diff
@@ -522,6 +522,62 @@ You can generate as many themes with whatever prefix you can think up!
 
 You could also [generate color utility classes using the generator](#utility-class-generator) that use these custom properties.
 
+## Namespaces
+Gorko supports 'namespacing' both the generated class and variable names by allowing you to specify a prefix.  This is done using the `namespace` map within `$gorko-config`.  The default `namespace` config looks like this:
```
Comment (Andy-set-studio) [inline readme.md:526]:
```suggestion
Gorko supports “namespacing” both the generated class and variable names by allowing you to specify a prefix.  This is done using the `namespace` map within `$gorko-config`.  The default `namespace` config looks like this:
```

Just a little nit here

File: readme.md [hunk 4]
```diff
@@ -522,6 +522,62 @@ You can generate as many themes with whatever prefix you can think up!
...
+  'namespace': (
+    'prefix': '',             // string
+    'classes': true,          // boolean or string
+    'vars': false             // boolean or string
+  )
+)
+````
+### Namespace Settings
+#### prefix
+Specifying a value for `prefix` will append that value to classes (by default) and css variables (opt-in).  You are responsible for appending any separating character, such as a dash or underscore.
```
Comment (Andy-set-studio) [inline readme.md]:
```suggestion
Specifying a value for `prefix` will append that value to classes (by default) and css variables (opt-in).  You are responsible for appending any separating character, such as a dash or underscore. For example, if your namespace is `my-app-`, you need to add the trailing `-` character in the `'prefix'` section.
```

**Review by Andy-set-studio — CHANGES_REQUESTED** [review]
This all looks great so far. I'll just give it a test drive locally.

One thing that would be great is instead of `vars`, we set that property to be `css-vars`, so it is consistent the the custom properties setup.

Really nice work though, @bytelabsco. Thank you so much

**Comment by Andy-set-studio** [comment]
Test drive was a great success. Really cool feature you've added here :)

**Comment by bytelabsco** [comment]
I went ahead and changed the 'vars' property to 'css-vars' to match the other configuration sections.

**Comment by Andy-set-studio** [comment]
Awesome, thank you. Let's get this merged!

**Review by Andy-set-studio — APPROVED** [review]

**Comment by Andy-set-studio** [comment]
Also, are you on Twitter so I can credit you with this release?

**Comment by motionsuggests** [comment]
This is super cool. Great work @bytelabsco !

**Pull request merged**